src/relax/analysis/struct_info_analysis.cc

```diff
@@ -1014,14 +1034,14 @@ class SymbolicVarCollector : public relax::ExprVisitor,
   }
 
   void VisitStructInfoExprField(const PrimExpr& expr) final {
-    if (mode_ == VisitMode::kMatchVarDef && expr->IsInstance<tir::VarNode>()) {
-      // populate symbolic var in first occurrence
-      const auto& var = Downcast<tir::Var>(expr);
-      if (defined_symbolic_var_.count(var) == 0) {
-        defined_symbolic_var_.insert(var);
+    if (mode_ >= VisitMode::kMatchVarDef) {
```

Comment:
Do we use this pattern elsewhere? I'm not personally very fond of it because it could create problems for adding more modes later, but I'm not against adding it if we do this already.

Reply:
The pattern of GT/LT enums is used a couple of places, most commonly in the `OpPatternKind` when checking if an operation is as permissible or more permissible than required for a transformation.  That said, I agree that this isn't the most readable, and it probably would be better to have separate visitors to collect defined variables and to collect free variables.